SublimeOnSaveBuild.py

```diff
@@ -15,10 +15,34 @@ def on_post_save(self, view):
         # Load filename filter. Again, a project level setting takes precedence.
         filename_filter = view.settings().get('filename_filter', global_settings.get('filename_filter', '*'))
 
+        # Check if we should automatically hide the build window 
+        auto_hide_build_window = view.settings().get('auto_hide_build_window', global_settings.get('auto_hide_build_window', True))
+
         if not should_build:
             return
 
         if not re.search(filename_filter, view.file_name()):
             return
 
+        # show the 'exec' view before building, so we can read from it afterwards
+        self.output_view = view.window().get_output_panel("exec")
+
         view.window().run_command('build')
+
+        if auto_hide_build_window:
+            self.num_polls = 0
+            # start polling for results every 100s
+            self.poll_for_results(view)
+
+    def poll_for_results(self, view):
+        build_finished = self.output_view.find('Finished', 0) != None
+
+        if build_finished:
+            errors = self.output_view.find('Error', 0)
+            if errors == None:
+                view.window().run_command("hide_panel", {"panel": "output.exec"})
```

Comment:
Actually I believe the `panel` argument to `hide_panel` is a no-op, unlike to its sibling `show_panel`. I just tried it out in ST2 and ST3 with the 'find' panel open; the above `hide_panel` call would still close the find panel even though the `output.exec` panel was specified. Seems https://sublimetext.userecho.com/topics/1930-add-panel-param-to-hide_panel-command/ is still open. @wbond, correct?